web-wallet/src/lib/components/__tests__/Transactions.spec.js

```diff
@@ -0,0 +1,158 @@
+import { cleanup, render } from "@testing-library/svelte";
+import { afterEach, describe, expect, it, vi } from "vitest";
+import Transactions from "../Transactions/Transactions.svelte";
+import { get } from "svelte/store";
+import { settingsStore } from "$lib/stores";
+
+import { transactions } from "$lib/mock-data";
+import { sortByHeightDesc } from "$lib/transactions";
+
+global.ResizeObserver = vi.fn().mockImplementation(() => ({
+	observe: vi.fn(),
+	unobserve: vi.fn(),
+	disconnect: vi.fn()
+}));
+
+vi.useFakeTimers();
+
+describe("Transactions", () => {
+	const transactionsPromise = Promise.resolve(transactions);
+	const emptyTransactionsPromise = Promise.resolve([]);
+	const blockExplorerBaseUrl = "https://explorer.dusk.network/transactions/transaction?id=";
+	const highestTransactionID = "tg874534tddsghdtst54hdfh795h";
+	const settings = get(settingsStore);
+	const language = settings.language;
+
+	afterEach(() => {
+		cleanup();
+	});
+
+	it("renders loading indicator before the promise has resolved", async () => {
+		const { getByRole } = render(
+			Transactions,
+			{ props: { language, items: transactionsPromise } }
```

Comment:
If you don't pass the target in the options object as we do elsewhere (and you're fine with having an extra `div` in the snapshots), you can just pass the props.
e.g.:
```js
const props = { language, items: transactionsPromise };
const { getByRole } = render(Transactions, props);
```

Reply:
Different tests use different props (eg. transactionsPromise, emptyTransactionsPromise, different limits value, etc.)

If I extract the props, as suggested above, it would be good to have a better naming (for example, if we were using the same props for all happy paths, we could have validProps) or something like that. Currently, each test uses different combination of props and I don't see a benefit from extracting the props as a constant as suggested above.

Reply:
> Different tests use different props (eg. transactionsPromise, emptyTransactionsPromise, different limits value, etc.)
> 
> If I extract the props, as suggested above, it would be good to have a better naming (for example, if we were using the same props for all happy paths, we could have validProps) or something like that. Currently, each test uses different combination of props and I don't see a benefit from extracting the props as a constant as suggested above.

The suggestion wasn't about extracting the props, but about passing the props directly, without an object having a "props" property:

```js
const { getByRole } = render(Transactions, { language, items: transactionsPromise });
```

Having said that I often have a `baseProps` value at the start of the test, inside the describe, with the most common props used during the test, and I modify those as needed inside the single test that needs a particular check.

Reply:
I guess it comes down to readability. If the props are not exported as a variable, then  just by looking at `render(Transactions, { language, items: transactionsPromise });` doesn't become immediately clear that language and items are props.

Reply:
My counterpoint would be that if in the future we add props to the component you would have to update every single test, while if you create a `baseProps` object at the beginning, inside a test you can just:

```js
const props = {
    ...baseProps,
    propIWantToChange: "foo"
}
```

Reply:
(which is also better for readability, in my opinion, as you see clearly what you're going to test differently from the usual "base props")